src/cursor/find_cursor.ts

```diff
@@ -101,7 +101,9 @@ export class FindCursor<TSchema = any> extends AbstractCursor<TSchema> {
         limit && limit > 0 && numReturned + batchSize > limit ? limit - numReturned : batchSize;
 
       if (batchSize <= 0) {
-        this.close().finally(() => callback());
+        this.close().finally(() =>
```

Comment:
Is this change just for consistency in return from `_getMore` or is there a reason it's needed?

Reply:
Yea, it is for result consistency, I may be able to revert it if I make sure the logic is nullish safe in next but the contract should have been either throw or result and when we make _getMore actually async that will be the case for the promise's resolution type. 

This code path also specifically claims it is for legacy servers which we don't support, I'm fairly certain this is only triggered by mock server tests.